tools/taskcluster/upload_apk_nimbledroid.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+This script executes upload_apk_nimbledroid.py and returns nonzero when the upload fails
+"""
+
+returnValue=$(python ./tools/taskcluster/upload_apk_nimbledroid.py)
+
+if [ ! -z "$returnValue" ]
+then
+    echo $returnValue
+    exit 1
+fi
```

Comment:
This script doesn't make any sense. You are checking whether the python script returned a non-zero exit code and if so return 1. A non-zero exit code is already enough to signal an error to taskcluster. No need to translate everything to 1 instead of the actual error.

Reply:
@pocmo but it seems that despite the print lines, TC thinks the job has passed, so probably python call doesn't 'return' anything to the shell? https://tools.taskcluster.net/groups/LUhigomASxyi6HH9jqgKow/tasks/Rd58QQ1jQGClOZoJBFwqqg/runs/0/logs/public%2Flogs%2Flive.log

Reply:
What the script does is to fetch the print statement from python, and convert it to shell exit value, since python doesn't seem to be able to provide return value to shell. (to my knowledge)